torch/distributed/_tensor/examples/torchrec_sharding_example.py

```diff
@@ -0,0 +1,173 @@
+    # over the 1-d mesh.
+    row_wise_sharding_placements = [Shard(0)]
+    # create a DTensor from the local shard
+    dtensor = DTensor.from_local(
+        local_shard, device_mesh, row_wise_sharding_placements, run_check=False
+    )
+    # display the DTensor's sharding
+    if rank == 0:  # the print result is identical on all ranks
+        print("DTensor's sharding be-like in row-wise sharding:")
+        visualize_sharding(dtensor)
+    # get the global tensor from the DTensor
+    dtensor_full = dtensor.full_tensor()  # torch.Tensor
+    # manually compose the global tensor from the local shards
+    global_tensor = torch.cat(local_shards, dim=0)
+    # we demonstrate that the DTensor constructed has the same
+    # global view as the actual global tensor
+    assert torch.equal(dtensor_full, global_tensor)
+
+    ###########################################################################
+    # example 2: transform DTensor into local_shards
+    # usage in TorchRec:
+    #   In ShardedEmbeddingCollection's load_state_dict pre hook
+    #   _pre_load_state_dict_hook, if the source param is a ShardedTensor
+    #   then we need to transform it into its local_shards.
+
+    # transform DTensor into local_shards
+    local_shard = dtensor.to_local()
+
+    # another case is that the source param is a torch.Tensor. In this case,
+    # the source param is the global tensor rather than shards so we need to
+    # splice the global tensor into local shards. This will be identical to
+    # existing code in TorchRec.
+    local_shard_shape_list = list(local_shard_shape)
+    local_shard_spliced = global_tensor[
+        local_shard_shape_list[0] * rank : local_shard_shape_list[0] * (rank + 1),
+        :,
+    ]
+    # the local shard obtained from both approaches should be identical
+    assert torch.equal(local_shard_spliced, local_shard_spliced)
+
+    ###########################################################################
+    # example 3: load state dict
+    # usage in TorchRec:
+    #   In case where the source param and the destination param are both
+    #   DTensors, we can directly call DTensor.copy_() to load the state.
+    src_dtensor = torch.distributed._tensor.ones(
+        emb_table_shape,
+        device_mesh=device_mesh,
+        placements=row_wise_sharding_placements,
+    )
+    dtensor.copy_(src_dtensor)
+    # these two DTensors should have the same global view after loading
+    assert torch.equal(dtensor.full_tensor(), src_dtensor.full_tensor())
+
+
+def run_example(rank, world_size, example_name):
+    # the dict that stores example code
+    name_to_example_code = {
+        "row-wise": run_torchrec_row_wise_sharding_example,
+    }
+    if example_name not in name_to_example_code:
+        print(f"example for {example_name} does not exist!")
+        return
+
+    # the example to run
+    example_func = name_to_example_code[example_name]
+
+    # set up world pg
+    os.environ["MASTER_ADDR"] = "localhost"
```

Comment:
if we use torchrun directly we don't need to setup those environments at all (from L138 - L146)